ts/parts/SvgRenderer.ts

```diff
@@ -2721,6 +2721,15 @@ extend((
             // compensate for inverted plot area
             transform;
 
+        // Update shadow when options change (#12091).
+        if (shadowOptions) {
+            if (JSON.stringify(this.oldShadowOptions) !== JSON.stringify(shadowOptions)) {
```

Comment:
`JSON.stringify` is not a safe solution when comparing objects. See demo: https://jsfiddle.net/BlackLabel/3k2j8ht9/